### PR TITLE
add multicuster cloud provider

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
## Command:
Make multicluster-provider-fake binary:
```
make multicluster-provider-fake
```

Make multicluster-provider-fake image:
```
make image-multicluster-provider-fake
```

Deploy multicluster-provider-fake deployment:
```
hack/deploy-provider.sh
```

Delete multicluster-provider-fake deployment :
```
kubectl --context karmada-host -n karmada-system delete deployments.apps multicluster-provider-fake
```
## Verify
<details>
<summary>mci.yaml</summary>

```yaml
apiVersion: networking.karmada.io/v1alpha1
kind: MultiClusterIngress
metadata:
  name: minimal-ingress
  annotations:
    nginx.ingress.kubernetes.io/rewrite-target: /
spec:
  rules:
  - http:
      paths:
      - path: /testpath
        pathType: Prefix
        backend:
          service:
            name: serve
            port:
              number: 80
```
</details>

<details>
<summary>application.yaml</summary>

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: serve
spec:
  replicas: 1
  selector:
    matchLabels:
      app: serve
  template:
    metadata:
      labels:
        app: serve
    spec:
      containers:
      - name: serve
        image: jeremyot/serve:0a40de8
        args:
        - "--message='hello from cluster member1 (Node: {{env \"NODE_NAME\"}} Pod: {{env \"POD_NAME\"}} Address: {{addr}})'"
        env:
          - name: NODE_NAME
            valueFrom:
              fieldRef:
                fieldPath: spec.nodeName
          - name: POD_NAME
            valueFrom:
              fieldRef:
                fieldPath: metadata.name
---
apiVersion: v1
kind: Service
metadata:
  name: serve
spec:
  ports:
  - port: 80
    targetPort: 8080
  selector:
    app: serve
---
apiVersion: policy.karmada.io/v1alpha1
kind: PropagationPolicy
metadata:
  name: mcs-workload
spec:
  resourceSelectors:
    - apiVersion: apps/v1
      kind: Deployment
      name: serve
    - apiVersion: v1
      kind: Service
      name: serve
  placement:
    clusterAffinity:
      clusterNames:
        - member1
        - member2
    replicaScheduling:
      replicaDivisionPreference: Weighted
      replicaSchedulingType: Divided
      weightPreference:
        staticWeightList:
          - targetCluster:
              clusterNames:
                - member1
            weight: 1
          - targetCluster:
              clusterNames:
                - member2
            weight: 1
```
</details>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - name: lint
+        run: hack/verify-staticcheck.sh
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.19
```
Comment (RainbowMango) [inline .github/workflows/ci.yml]:
```suggestion
        go-version: 1.19.5
```

File: artifacts/deploy/ingressclass-fake.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: networking.k8s.io/v1
+kind: IngressClass
+metadata:
+  annotations:
+    ingressclass.kubernetes.io/is-default-class: "true"
+  name: fake
+spec:
+  controller: karmada.io/fake
```
Comment (RainbowMango) [inline artifacts/deploy/ingressclass-fake.yaml]: end line.

File: pkg/controllers/crdsynchronizer/crd_synchronizer.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// ControllerName is the controller name that will be used when reporting events.
+const ControllerName = "crd-synchronizer"
```
Comment (RainbowMango) [inline pkg/controllers/crdsynchronizer/crd_synchronizer.go]: We might consider a more straightforward name.

File: pkg/version/release.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
+package version
```
Comment (RainbowMango) [inline pkg/version/release.go]: Why copy this package? Can we import it from karmada repo?

File: pkg/controllers/crdsynchronizer/crd_synchronizer.go (hunk 3 above)
Comment (XiShanYongYe-Chang) [reply pkg/controllers/crdsynchronizer/crd_synchronizer.go]: How about `serviceexport-crd-synchronizer`?

File: pkg/version/release.go (hunk 4 above)
Comment (XiShanYongYe-Chang) [reply pkg/version/release.go]: Import it form karmada repo.

File: pkg/controllers/crdsynchronizer/crd_synchronizer.go (hunk 3 above)
Comment (XiShanYongYe-Chang) [reply pkg/controllers/crdsynchronizer/crd_synchronizer.go]: How about `crd-installer-controller`? Considering that other crd files may be installed in the future, they can be processed in this controller.

Comment (RainbowMango) [reply pkg/controllers/crdsynchronizer/crd_synchronizer.go]: ok, that makes sense.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Comment by RainbowMango** [comment]
/lgtm

Test prow bot...

**Pull request merged**